oak_services/proto/http_encap.proto

```diff
@@ -27,15 +27,22 @@ message HttpRequest {
   // The body of the request.
   bytes body = 3;
   // The HTTP request headers.
-  map<string, bytes> headers = 4;
+  map<string, HeaderValue> headers = 4;
```

Comment:
Would it be simpler to just have a `Header` message that encapsulates name and value, and may be repeated? That seems to map more closely to HTTP semantics, in which repeated headers with the same name are not really related to each other. Also it seems simpler to create and pass around, and would presumably map nicely to something like https://docs.rs/hyper/0.13.8/hyper/header/struct.HeaderMap.html#impl-FromIterator%3C(HeaderName%2C%20T)%3E or other multimap-style collections in Rust. WDYT?

Reply:
We are mostly converting the headers to hyper headers. For that I don't see a huge difference between the two types. But we also directly lookup in the headers in `get_oak_label`. For that it is easier to have a map. As far as I know, multimaps are just wrappers around Maps. But I don't think `prost` will create a multimap for `repeated Header`, so in my opinion `map<string, HeaderValue>` is a more suitable type.